.travis.yml

```diff
@@ -1,41 +1,160 @@
+    # - NIX_PATH="nixpkgs=https://github.com/nixos/nixpkgs/archive/$rev.tar.gz"
+    # NOTE: Project/binary name
+    - name=hnix
+    # NOTE: Used in the `generateOptparseApplicativeCompletions = true`
+    - pkgName='haskellPackages.hnix'
+    # NOTE: Enable all our tests in cabal
+    - ALL_TESTS=yes
+    # NOTE: Turn all warn into err with {-Wall,-Werror}
+    - failOnAllWarnings=false
+    # NOTE: checkUnusedPackages: is `failOnAllWarnings` + `cabal sdist` to ensure all needed files are listed in the Cabal file. Uses `packunused` or GHC internals. Adds a post-build check to verify that dependencies declared in the cabal file are actually used. The first attrset argument can be used to configure the strictness of this check and a list of ignored package names that would otherwise cause false alarms.
+    - checkUnusedPackages=false
+    # NOTE: Generation and installation of a coverage report.
+    # See https://wiki.haskell.org/Haskell_program_coverage
+    - doCoverage=false
+    # NOTE: Generation and installation of haddock API documentation
+    - doHaddock=false
+    # NOTE: Escape the version bounds from the cabal file. You may want to avoid this function.
+    - doJailbreak=false
+    # NOTE: Disables Nix dependency checking, compilation and execution of test suites listed in the package description file.
+    - doCheck=true
+    # NOTE: doBenchmark: Dependency checking + compilation and execution for benchmarks listed in the package description file.
+    - doBenchmark=false
+    - enableExecutableProfiling=false
+    - enableLibraryProfiling=false
+    # NOTE: Build a source distribution tarball instead of using the source files directly. The effect is that the package is built as if it were published on hackage. This can be used as a test for the source distribution, assuming the build fails when packaging mistakes are in the cabal file.
+    - buildFromSdist=false
+    # NOTE: Build the package in a strict way to uncover potential problems. This includes buildFromSdist and failOnAllWarnings.
+    #  2020-05-26: NOTE: Currently HNix not able to pass Strict
+    - buildStrictly=false
+    # NOTE: Disable core optimizations, significantly speeds up the build
+    - disableOptimization=true
+    - buildStackProject=false
+    # NOTE: Modify a Haskell package to add shell completion scripts for the given executable produced by it. These completion scripts will be picked up automatically if the resulting derivation is installed, e.g. by `nix-env -i`.
+    # Invocation:
+    #   generateOptparseApplicativeCompletions command pkg
+    #
+    #   command: name of an executable
+    #       pkg: Haskell package that builds the executables
+    - generateOptparseApplicativeCompletion=false
+    # NOTE: Don't fail at configure time if there are multiple versions of the same package in the (recursive) dependencies of the package being built. Will delay failures, if any, to compile time.
+    - allowInconsistentDependencies=false
```

Comment:
That sounds like cool, but possibly dangerous(?) feature! It might be useful to avoid disabling the `executable` in https://github.com/haskell-nix/hnix/pull/540.

Reply:
That is an option that is turned off but provided.

There is a lot of dangerous options provided. Since Nix is pure, restrictive, and idealistic and still depends on Cabal descriptions.

That they are there does not mean that we need to enable them.